### PR TITLE
Fix AO2Ds in case of negative first daughter 

### PR DESCRIPTION
**Pull request by mpuccio**
@jgrosseo

### CODE REVIEW

**Review by jgrosseo — DISMISSED** [review]

**Comment by jgrosseo** [comment]
@sawenzel This check will most likely break all ongoing productions (from @mpuccio investigation), it is still the right thing to do...

**Comment by chiarazampolli** [comment]
Hi @jgrosseo , 

Do you mean that from the moment when it is merged, all jobs will fail, not passing the test?
See also my comment in https://alice.its.cern.ch/jira/browse/O2-2820

Chiara

**Comment by jgrosseo** [comment]
This concerns only MC. Probably not all but quite some will fail. At least Max found this in a number of AODs. I do not have answers to your other questions in the ticket...

**Comment by sawenzel** [comment]
I was always in favour of having such checks but rather on the analysis level. (Doing analysis regression tests on the alibi-nightlies). I am not in favour of introducing this now here, as it will

File: Detectors/AOD/src/AODProducerWorkflowSpec.cxx [hunk 1]
```diff
@@ -728,6 +728,9 @@ void AODProducerWorkflowDPL::fillMCParticlesTable(o2::steer::MCKinematicsReader&
       if (daughters[0] > daughters[1]) {
         std::swap(daughters[0], daughters[1]);
       }
+      if (daughters[0] < 0 && daughters[1] >= 0) {
+        LOG(fatal) << "Discordant daughters indices (" << daughters[0] << "," << daughters[1] << "): the AO2D would be nonconforming to the data model, aborting.";
+      }
```
Comment (sawenzel) [inline Detectors/AOD/src/AODProducerWorkflowSpec.cxx]: so just to understand: if daughter index 0 is negative (== no daughter) we should not have a daughter in index 1 (second daughter)?

It appears to me that the 2 lines above will already cause such misbehaviour. We seem to swap the index whenever daughters[0] > daughters[1] which happens in particular when daughters[1] == -1. @shahor02 : Can you comment on this?

**Comment by mpuccio** [comment]
Hello everyone, after discussing with @sawenzel we came to the conclusion that these cases might be due to pruned particles in the kinematics. We will have a look in details to check that this is the case and that no physics is harmed (there is already quite some certainty about this, as the `keepPhysics` function in Stack keeps primaries and their offsprings).
The current solution will make the AO2D conforming to the data model and will not disrupt current productions.

**Pull request merged**